### PR TITLE
OperatorsSpacesFixer enhancements

### PR DESCRIPTION
**Pull request by gharlan**
http://symfony.com/doc/current/contributing/code/standards.html#structure:

> * Add a single space around binary operators (==, &&, ...), with the exception of the concatenation (.) operator;
> * Place unary operators (!, --, ...) adjacent to the affected variable;

So I made these enhancements for the `OperatorsSpacesFixer`:

* added unary operator checks (`++`, `--`, `+`, `-`, `!`, `&`, `~`, `@`, `...`)
* added missing binary operators (`+`, `-`, `*`, `/`, `%`, `<`, `>`, `&`, `|`, `^`, `**`, `**=`, `<=>`, `??`)

### CODE REVIEW

File: Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php [hunk 1]
```diff
@@ -159,9 +226,107 @@ public function provideCases()
...
+        $this->makeTest($expected, $input);
+    }
+
+    public function provideCases70()
+    {
+        return array(
+            array(
+                '<?php $a <=> $b;',
+                '<?php $a<=>$b;',
+            ),
```
Comment (gharlan) [inline Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php]: hmm, why is this failing with PHP 7: https://travis-ci.org/FriendsOfPHP/PHP-CS-Fixer/jobs/54506137

File: Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php [hunk 2]
```diff
@@ -30,6 +31,72 @@ public function provideCases()
...
+                '<?php $a = !$b;',
+                '<?php $a = ! $b;',
+            ),
+            array(
+                '<?php func(+$a, -2,-$b);',
+                '<?php func(+ $a, - 2,- $b);',
+            ),
+            array(
+                '<?php $a = &$b;',
+                '<?php $a = & $b;',
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php]: IMO this should be two fixers.
One for binary ops., one for unary ops.
I prefer `&` sticked to variable, but not all do.

File: Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php [hunk 3]
```diff
@@ -30,6 +31,72 @@ public function provideCases()
...
+            array(
+                '<?php $a = !$b;',
+                '<?php $a = ! $b;',
+            ),
+            array(
+                '<?php func(+$a, -2,-$b);',
+                '<?php func(+ $a, - 2,- $b);',
+            ),
+            array(
+                '<?php $a = &$b;',
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php]: Please add a cases:
- `<?php $a=&$b;`
- `<?php function & foo(){}`

File: Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php (hunk 2 above)
Comment (gharlan) [inline Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php]: Ok, makes sense. 
So I will add a `UnaryOperatorsSpacesFixer`?
And `OperatorsSpacesFixer` (bc name) will fix binary operators only.

File: Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php (hunk 1 above)
Comment (gharlan) [inline Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php]: https://github.com/php/php-src/commit/0a4a11b73ae32b31810451d1f7e8719ca0a503db
this commit is created 3 days ago, but travis is using a 7.0 version built 4 days ago. I guess that's the reason. ;)

File: Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php (hunk 2 above)
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php]: ok

File: Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php (hunk 1 above)
Comment (gharlan) [inline Symfony/CS/Tests/Fixer/Symfony/OperatorsSpacesFixerTest.php]: it works now: https://travis-ci.org/FriendsOfPHP/PHP-CS-Fixer/jobs/55212635

**Comment by gharlan** [comment]
this fixes #81

**Comment by keradus** [comment]
could you rebase please on current 1.6 please?

**Comment by gharlan** [comment]
done

**Comment by keradus** [comment]
Thanks !

Any chances to create tests dedicated for new `Tokens` methods ?

**Comment by gharlan** [comment]
> Any chances to create tests dedicated for new Tokens methods ?

done

**Comment by GrahamCampbell** [comment]
:+1: :ship:

**Comment by gharlan** [comment]
(added `@` to unary predecessor operators)

**Comment by keradus** [comment]
Regression on Symfony repo has been made.

**Comment by keradus** [comment]
Thank you @gharlan.

**Pull request merged**